src/sites/twitch-twilight/modules/directory/game.js

```diff
@@ -98,157 +59,15 @@ export default class Game extends SiteModule {
-		if (hiddenBodyCard !== null) hiddenBodyCard.classList.remove('hide');
-
-		const ffzChannelData = card.querySelector('.ffz-channel-data');
-		if (ffzChannelData !== null) ffzChannelData.remove();
-
-		const channelAvatar = card.querySelector('.channel-avatar');
-		if (channelAvatar !== null) channelAvatar.remove();
-
-		const hiddenThumbnails = this.settings.provider.get('directory.game.hidden-thumbnails') || [];
-		const thumbnailBlocked = hiddenThumbnails.includes(inst.props.directoryName);
-
-		if (thumbnailBlocked) {
-			card.classList.add('ffz-thumbnail-hidden');
-		} else {
-			card.classList.remove('ffz-thumbnail-hidden');
-		}
-
-		if (inst.props.streamNode.viewersCount.profileImageURL) {
-			const avatarSetting = this.settings.get('directory.following.show-channel-avatar');
-			if (avatarSetting === 1) {
-				const cardDiv = card.querySelector('.tw-card-body');
-				const modifiedDiv = e('div', {
-					innerHTML: cardDiv.innerHTML
-				});
-
-				const avatarDiv = e('a', {
-					className: 'channel-avatar',
-					href: `/${inst.props.streamNode.broadcaster.login}`,
-					style: 'margin-right: 8px; min-width: 4rem; margin-top: 0.5rem;',
-					onclick: event => {
-						event.preventDefault();
-						event.stopPropagation();
-
-						this.router.navigate('user', { userName: inst.props.streamNode.broadcaster.login});
-					}
-				}, e('img', {
-					title: inst.props.streamNode.broadcaster.displayName,
-					src: inst.props.streamNode.viewersCount.profileImageURL,
-					style: 'height: 4rem;'
-				}));
-
-				const cardDivParent = cardDiv.parentElement;
-
-				if (cardDivParent.querySelector('.ffz-channel-data') === null) {
-					cardDiv.classList.add('hide');
-
-					const newCardDiv = e('div', 'ffz-channel-data flex flex-nowrap', [
-						avatarDiv, modifiedDiv
-					]);
-					cardDivParent.appendChild(newCardDiv);
-				}
-			} else if (avatarSetting === 2 || avatarSetting === 3) {
-				const avatarElement = e('a', {
-					className: 'channel-avatar',
-					href: `/${inst.props.streamNode.broadcaster.login}`,
-					onclick: event => {
-						event.preventDefault();
-						event.stopPropagation();
-
-						this.router.navigate('user', { userName: inst.props.streamNode.broadcaster.login});
-					}
-				}, e('div', 'live-channel-card__boxart bottom-0 absolute',
-					e('figure', 'tw-aspect tw-aspect--align-top',
-						e('img', {
-							title: inst.props.streamNode.broadcaster.displayName,
-							src: inst.props.streamNode.viewersCount.profileImageURL
-						})
-					)
-				)
-				);
-
-				const divToAppend = card.querySelector('figure.tw-aspect');
-				if (divToAppend.querySelector('.channel-avatar') === null) divToAppend.appendChild(avatarElement);
-			}
-		}
+		if (this.router.current.name !== 'dir-game-index') return;
+		
+		this.parent.updateUptime(inst, 'props.streamNode.viewersCount.createdAt', '.tw-thumbnail-card .tw-card-img');
+		this.parent.addCardAvatar(inst, 'props.streamNode.viewersCount.createdAt', '.tw-thumbnail-card');
```

Comment:
Maybe instead of having separate route logic for these, you can make one method that checks what kind of card it is so it knows what selector to pass into these?